osclib/check_duplicate_binaries_command.py

```diff
@@ -13,30 +16,41 @@ def __init__(self, api):
     def perform(self, save=False):
         duplicates = {}
         for arch in sorted(target_archs(self.api.apiurl, self.api.project), reverse=True):
-            print('arch {}'.format(arch))
+	    url = self.api.makeurl(['build', self.api.project, 'standard', arch], { 'view': 'binaryversions' })
+	    try:
+		data = http_GET(url)
+	    except urllib2.HTTPError:
+		return
+	    root = ET.parse(data).getroot()
 
             binaries = {}
-            duplicates[arch] = {}
-            for package in package_list(self.api.apiurl, self.api.project):
-                for binary in get_binarylist(self.api.apiurl, self.api.project,
-                                             'standard', arch, package):
+	    for packagenode in root.findall('.//binaryversionlist'):
+		package = packagenode.get('package')
+                for binarynode in packagenode.findall('binary'):
+		    binary = binarynode.get('name')
                     # StagingAPI.fileinfo_ext(), but requires lots of calls.
                     match = re.match(r'(.*)-([^-]+)-([^-]+)\.([^-\.]+)\.rpm', binary)
-                    if not match or match.group(4) == 'src': continue
+		    if not match:
+			continue
+		    parch = match.group(4)
+		    if parch in ('src', 'nosrc'):
+			continue
 
-                    name = match.group(1)
-                    if name in binaries:
-                        print('DUPLICATE', package, binaries[name], name)
+		    # XXX: we assume that proper export filters for
+		    # the architecures exist. Examples are glibc and
+		    # syslinux.
```

Comment:
Not sure what this comment is trying to convey.

In general a `noarch` does duplicate a `x86_64` and the same should be true of other arch combinations. As such restricting binary check to `parch` seems wrong. `glibc` being a special case since it is allowed to have both `i586` and `i686`. As such it seems to make sense to provide a whitelist or other method for filtering such cases out.